oak_runtime/src/node/http/tests.rs

```diff
@@ -197,35 +221,39 @@ fn create_communication_channel(runtime: &RuntimeProxy) -> (oak_abi::Handle, oak
     (init_receiver, invocation_receiver)
 }
 
-fn oak_node_simulator(runtime: &RuntimeProxy, invocation_receiver: oak_abi::Handle) {
-    // Get invocation message that contains the response_writer handle.
-    let invocation_receiver = Receiver::<HttpInvocation>::new(ReadHandle {
-        handle: invocation_receiver,
-    });
-    let invocation = invocation_receiver.receive(runtime).unwrap();
-    let resp = HttpResponse {
-        body: vec![],
-        status: http::status::StatusCode::OK.as_u16() as i32,
-        headers: hashmap! {},
-    };
-    invocation
-        .sender
-        .expect("Empty sender on invocation.")
-        .send(resp, runtime)
-        .unwrap();
+// Simulate an Oak node that responds to the specified number of requests. Since the Oak node blocks
+// until a new request is received, we need to know the number of requests that the test sends
+// beforehand. Alternatively, the node simulator could listen for incoming requests in an infinite
+// loop and have a termination mechanism that could be signalled from outside. However, specifying
```

Comment:
Since it's reading from a channel, why not just read from the channel until the channel is exhausted, which is what we normally do in the actual nodes? Then the signal to terminate would be closing the channel.

Reply:
I don't know what signals the exhaustion of the channel. Do you have a code example? 
The documentation of `receive` only says "**Waits, reads and decodes a message from the [`Receiver::handle`].**". This documentation must be extended with details like the channel exhaustion, error conditions, and example usage.

Reply:
I don't have the details at hand, but roughly something like this:

https://github.com/project-oak/oak/blob/984b3660a9bd9e551547d40c5314517011c22f36/oak_runtime/src/node/logger.rs#L58-L91

Reply:
I tried with an infinite loop that terminates in case of any error, but it blocks for ever. What is signalling the end of the channel?

Reply:
It should be signalled by closing the channel on the other side, or the runtime terminating. Maybe neither of these things is happening in tests though.

Reply:
I think the problem is that the nodes in the test are not created by the runtime (not via `Runtime::`). I'll see if I can fix that, in another PR.